mentoring/v1/upgrade.py

```diff
@@ -0,0 +1,168 @@
+
+    # Was block already published?
+    parent = block.get_parent()
+    parent_was_published = not store.has_changes(parent)
+
+    # If the block has a url_name attribute that doesn't match Studio's url_name, fix that:
+    delete_on_success = None
+    if "url_name" in root.attrib:
+        url_name = root.attrib.pop("url_name")
+        if block.url_name != url_name:
+            print(" ➔ This block has two conflicting url_name values set. Attempting to fix...")
+            # Fix the url_name by replacing the block with a blank block with the correct url_name
+            parent_children = parent.children
+            old_usage_id = block.location
+            index = parent_children.index(old_usage_id)
+            try:
+                new_block = store.create_item(
+                    user_id=None,
+                    course_key=block.location.course_key,
+                    block_type="mentoring",
+                    block_id=url_name,
+                    fields={"xml_content": xml_content_str},
+                )
+                delete_on_success = block
+                parent_children[index] = new_block.location
+                parent.save()
+                store.update_item(parent, user_id=None)
+                block = new_block
+                print(" ➔ url_name changed to {}".format(url_name))
+                # Now we've fixed the block's url_name but in doing so we've disrupted the student data.
+                # Migrate it now:
+                student_data = StudentModule.objects.filter(module_state_key=old_usage_id)
+                num_entries = student_data.count()
+                if num_entries > 0:
+                    print(" ➔ Migrating {} student records to new url_name".format(num_entries))
+                    student_data.update(module_state_key=new_block.location)
+            except DuplicateItemError:
+                print(
+                    "\n WARNING: The block with url_name '{}' doesn't match "
+                    "the real url_name '{}' and auto repair failed.\n".format(
+                        url_name, block.url_name
+                    )
+                )
+
+    # Replace block with the new version and the new children:
+    studio_update_from_node(block, root)
+
+    if delete_on_success:
+        store.delete_item(delete_on_success.location, user_id=None)
+    if parent_was_published:
+        store.publish(parent.location, user_id=None)
+
+
+if __name__ == '__main__':
+    # Disable some distracting overly-verbose warnings that we don't need:
+    for noisy_module in ('edx.modulestore', 'elasticsearch', 'urllib3.connectionpool'):
+        logging.getLogger(noisy_module).setLevel(logging.ERROR)
+    from opaque_keys import InvalidKeyError
+    from opaque_keys.edx.keys import CourseKey
+    from xmodule.modulestore.django import modulestore
+
+    print("┏━━━━━━━━━━━━━━━━━━━━━━━━━━┓")
+    print("┃ Mentoring Upgrade Script ┃")
+    print("┗━━━━━━━━━━━━━━━━━━━━━━━━━━┛")
```

Comment:
FYI: Loving the use of unicode. You could use `textwrap.dedent` to only have to call print once.